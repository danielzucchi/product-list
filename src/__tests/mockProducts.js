const mockProducts = [
  {
    id: '1',
    date: '2020-11-11T03:23:31.984Z',
    name: 'Rowena Jerde',
    img: 'https://s3.amazonaws.com/uifaces/faces/twitter/a_harris88/128.jpg',
    sold: false,
    price: '51.00',
    brand: 'Fantastic Concrete Computer',
    description:
      "You can't copy the capacitor without generating the digital SAS hard drive!",
    seller: 'Frederic',
  },
  {
    id: '2',
    date: '2020-11-11T03:25:47.608Z',
    name: 'Walter Nicolas',
    img: 'https://s3.amazonaws.com/uifaces/faces/twitter/uxalex/128.jpg',
    sold: true,
    price: '348.00',
    brand: 'Handcrafted Metal Car',
    description:
      "I'll override the open-source HTTP protocol, that should interface the COM program!",
    seller: 'Lonnie',
  },
  {
    id: '3',
    date: '2020-11-11T08:07:26.985Z',
    name: 'Justina Heaney',
    img: 'https://s3.amazonaws.com/uifaces/faces/twitter/juangomezw/128.jpg',
    sold: false,
    price: '461.00',
    brand: 'Unbranded Frozen Pizza',
    description:
      'Try to synthesize the PNG monitor, maybe it will parse the primary pixel!',
    seller: 'Calista',
  },
  {
    id: '4',
    date: '2020-11-10T17:50:08.753Z',
    name: 'Mrs. Elizabeth Tremblay',
    img: 'https://s3.amazonaws.com/uifaces/faces/twitter/operatino/128.jpg',
    sold: false,
    price: '71.00',
    brand: 'Handcrafted Rubber Chair',
    description:
      "overriding the transmitter won't do anything, we need to back up the primary HTTP driver!",
    seller: 'Kirk',
  },
]

export default mockProducts
