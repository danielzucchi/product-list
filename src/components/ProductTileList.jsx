import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import ProductTile from './ProductTile'

const ProductListContainer = styled.div`
  margin: 30px auto;
  width: 100%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: flex-start;
  justify-content: space-around;
`

const ProductTileList = ({
  products,
  hideSold,
  likedProducts,
  toggleLiked,
}) => {
  const filteredSoldProducts = products.filter(
    (product) => product.sold === false,
  )
  const productsToDisplay = hideSold ? filteredSoldProducts : products

  return (
    <ProductListContainer>
      {productsToDisplay.map(({ id, name, img, sold, price, brand }) => {
        const isLiked = likedProducts.includes(id)

        return (
          <li style={{ listStyle: 'none' }} key={id}>
            <ProductTile
              id={id}
              name={name}
              img={img}
              sold={sold}
              price={price}
              brand={brand}
              isLiked={isLiked}
              toggleLiked={toggleLiked}
            />
          </li>
        )
      })}
    </ProductListContainer>
  )
}

ProductTileList.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      date: PropTypes.string,
      name: PropTypes.string,
      img: PropTypes.string,
      sold: PropTypes.bool,
      price: PropTypes.string,
      brand: PropTypes.string,
      description: PropTypes.string,
      seller: PropTypes.string,
    }),
  ).isRequired,
  hideSold: PropTypes.bool,
  likedProducts: PropTypes.arrayOf(PropTypes.string).isRequired,
  toggleLiked: PropTypes.func.isRequired,
}

ProductTileList.defaultProps = {
  hideSold: false,
}

export default ProductTileList
