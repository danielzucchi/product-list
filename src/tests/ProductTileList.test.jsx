import React from 'react'
import { shallow } from 'enzyme'
import ProductTileList from '../components/ProductTileList'
import mockProducts from './mockProducts'

describe('ProducTileList test', () => {
  it('when all props are presentmatches snapshot', () => {
    const wrapper = shallow(
      <ProductTileList
        hideSold={false}
        products={mockProducts}
        likedProducts={['2', '3']}
        toggleLiked={() => jest.fn()}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('when hideSold products is off, show sold products', () => {
    const wrapper = shallow(
      <ProductTileList
        hideSold={false}
        products={mockProducts}
        likedProducts={['2', '3']}
        toggleLiked={() => jest.fn()}
      />,
    )
    const list = wrapper.findWhere((node) => node.name() === 'ProductTile')
    expect(list).toHaveLength(4)
  })

  it('when hideSold products is on, hides sold products', () => {
    const wrapper = shallow(
      <ProductTileList
        hideSold
        products={mockProducts}
        likedProducts={['2', '3']}
        toggleLiked={() => jest.fn()}
      />,
    )
    const list = wrapper.findWhere((node) => node.name() === 'ProductTile')
    expect(list).toHaveLength(3)
  })
})
