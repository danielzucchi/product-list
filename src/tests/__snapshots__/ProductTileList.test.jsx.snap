// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProducTileList test when all props are presentmatches snapshot 1`] = `
<styled.div>
  <li
    key="1"
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <ProductTile
      brand="Fantastic Concrete Computer"
      id="1"
      img="https://s3.amazonaws.com/uifaces/faces/twitter/a_harris88/128.jpg"
      isLiked={false}
      name="Rowena Jerde"
      price="51.00"
      sold={false}
      toggleLiked={[Function]}
    />
  </li>
  <li
    key="2"
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <ProductTile
      brand="Handcrafted Metal Car"
      id="2"
      img="https://s3.amazonaws.com/uifaces/faces/twitter/uxalex/128.jpg"
      isLiked={true}
      name="Walter Nicolas"
      price="348.00"
      sold={true}
      toggleLiked={[Function]}
    />
  </li>
  <li
    key="3"
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <ProductTile
      brand="Unbranded Frozen Pizza"
      id="3"
      img="https://s3.amazonaws.com/uifaces/faces/twitter/juangomezw/128.jpg"
      isLiked={true}
      name="Justina Heaney"
      price="461.00"
      sold={false}
      toggleLiked={[Function]}
    />
  </li>
  <li
    key="4"
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <ProductTile
      brand="Handcrafted Rubber Chair"
      id="4"
      img="https://s3.amazonaws.com/uifaces/faces/twitter/operatino/128.jpg"
      isLiked={false}
      name="Mrs. Elizabeth Tremblay"
      price="71.00"
      sold={false}
      toggleLiked={[Function]}
    />
  </li>
</styled.div>
`;
